### ISSUE TITLE
Errors while building under Archlinux

### ISSUE THREAD

**Issue by makefu** [DataSoft/Honeyd#1]
Hi I tried your fork today but stumbled across multiple errors when running "make"
1. encode_int function is already declared in libevent2
   see: https://bugs.gentoo.org/show_bug.cgi?id=333099
   renaming encode_int to encode__int helped (as stated in the bugreport)
2. in ui.c the compiling errors with "dereferencing pointer to incomplete type":
   make  all-recursive
   make[1]: Entering directory `/home/makefu/r/Honeyd/honeyd'
   Making all in .
   make[2]: Entering directory`/home/makefu/r/Honeyd/honeyd'
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c honeyd.c
   In file included from stats.h:36:0,
                from honeyd.c:98:
   ./compat/sha1.h:23:3: warning: ‘**bounded**’ attribute directive ignored [-Wattributes]
   ./compat/sha1.h:23:3: warning: ‘**bounded**’ attribute directive ignored [-Wattributes]
   ./compat/sha1.h:26:3: warning: ‘**bounded**’ attribute directive ignored [-Wattributes]
   ./compat/sha1.h:28:3: warning: ‘**bounded**’ attribute directive ignored [-Wattributes]
   ./compat/sha1.h:30:3: warning: ‘**bounded**’ attribute directive ignored [-Wattributes]
   ./compat/sha1.h:32:3: warning: ‘**bounded**’ attribute directive ignored [-Wattributes]
   ./compat/sha1.h:35:3: warning: ‘**bounded**’ attribute directive ignored [-Wattributes]
   ./compat/sha1.h:35:3: warning: ‘**bounded**’ attribute directive ignored [-Wattributes]
   honeyd.c: In function ‘tcp_send’:
   honeyd.c:1432:3: warning: implicit declaration of function ‘tcp_personality_test’ [-Wimplicit-function-declaration]
   honeyd.c:1432:12: warning: assignment makes pointer from integer without a cast [enabled by default]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c command.c
   command.c: In function ‘cmd_trigger_read’:
   command.c:87:3: warning: implicit declaration of function ‘asprintf’ [-Wimplicit-function-declaration]
   test -f parse.c || /bin/sh ./ylwrap parse.y y.tab.c parse.c y.tab.h parse.h y.output parse.output -- bison -y -d 
   updating parse.h
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c parse.c
   parse.y: In function ‘hydparse’:
   parse.y:1071:3: warning: implicit declaration of function ‘strptime’ [-Wimplicit-function-declaration]
   parse.y:1071:43: warning: comparison between pointer and integer [enabled by default]
   parse.y:1073:50: warning: comparison between pointer and integer [enabled by default]
   parse.y: In function ‘hyderror’:
   parse.y:1168:3: warning: implicit declaration of function ‘vasprintf’ [-Wimplicit-function-declaration]
   test -f lex.c || /bin/sh ./ylwrap lex.l lex.hyd.c lex.c -- flex -Phyd 
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c lex.c
   lex.c:1814:17: warning: ‘yyunput’ defined but not used [-Wunused-function]
   lex.c:1855:16: warning: ‘input’ defined but not used [-Wunused-function]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c config.c
   config.c: In function ‘template_print’:
   config.c:914:2: warning: format ‘%lx’ expects argument of type ‘long unsigned int’, but argument 3 has type ‘uint32_t’ [-Wformat]
   config.c: In function ‘template_test_parse_error’:
   config.c:1003:12: warning: pointer targets in initialization differ in signedness [-Wpointer-sign]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c personality.c
   personality.c: In function ‘tcp_personality_seq’:
   personality.c:658:6: warning: variable ‘slowhz’ set but not used [-Wunused-but-set-variable]
   personality.c: In function ‘tcp_personality_options’:
   personality.c:843:21: warning: unused variable ‘tv’ [-Wunused-variable]
   personality.c: At top level:
   personality.c:2608:1: warning: ‘print_xprobe_struct’ defined but not used [-Wunused-function]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c util.c
   util.c: In function ‘fdshare_close’:
   util.c:564:2: warning: implicit declaration of function ‘asprintf’ [-Wimplicit-function-declaration]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c ipfrag.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c router.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c tcp.c
   tcp.c: In function ‘cmd_tcp_eread’:
   tcp.c:144:2: warning: implicit declaration of function ‘asprintf’ [-Wimplicit-function-declaration]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c udp.c
   udp.c: In function ‘cmd_udp_eread’:
   udp.c:78:2: warning: implicit declaration of function ‘asprintf’ [-Wimplicit-function-declaration]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c xprobe_assoc.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c log.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c fdpass.c
   fdpass.c: In function ‘send_fd’:
   fdpass.c:66:2: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
   fdpass.c: In function ‘receive_fd’:
   fdpass.c:148:2: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c atomicio.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c subsystem.c
   subsystem.c: In function ‘subsystem_listen’:
   subsystem.c:312:2: warning: implicit declaration of function ‘asprintf’ [-Wimplicit-function-declaration]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c hooks.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c plugins.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c plugins_config.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c pool.c
   296 addr_unmarshal(struct addr\* addr, struct evbuffer _evbuf)
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c interface.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c arp.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c gre.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c network.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c pfctl_osfp.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c pf_osfp.c
   pf_osfp.c: In function ‘pf_osfp_fingerprint_hdr’:
   pf_osfp.c:88:7: warning: pointer targets in assignment differ in signedness [-Wpointer-sign]
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c condition.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c osfp.c
   gcc -DHAVE_CONFIG_H -I. -I./ -I./compat -I/usr/include/pcap -I/usr/include     -O2 -Wall -g -DPATH_HONEYDINCLUDE="\"/usr/local/include/honeyd\"" -DPATH_HONEYDDATA="\"/usr/local/share/honeyd\"" -DPATH_HONEYDLIB="\"/usr/local/lib/honeyd\"" -DHONEYD_PLUGINS_DECLARE="" -DHONEYD_PLUGINS="" -DPATH_RRDTOOL="\"\"" -c ui.c
   ui.c: In function ‘ui_write_prompt’:
   ui.c:132:16: warning: pointer targets in initialization differ in signedness [-Wpointer-sign]
   ui.c:134:2: warning: pointer targets in passing argument 1 of ‘strlen’ differ in signedness [-Wpointer-sign]
   /usr/include/string.h:399:15: note: expected ‘const char *’ but argument is of type ‘u_char *’
   ui.c: In function ‘ui_buffer_prompt’:
   ui.c:143:16: warning: pointer targets in initialization differ in signedness [-Wpointer-sign]
   ui.c:145:2: warning: pointer targets in passing argument 1 of ‘strlen’ differ in signedness [-Wpointer-sign]
   /usr/include/string.h:399:15: note: expected ‘const char *’ but argument is of type ‘u_char *’
   ui.c: In function ‘ui_writer’:
   ui.c:249:22: error: dereferencing pointer to incomplete type
   ui.c:249:38: error: dereferencing pointer to incomplete type
   ui.c:263:12: error: dereferencing pointer to incomplete type
   ui.c: In function ‘ui_handler’:
   ui.c:280:11: error: dereferencing pointer to incomplete type
   ui.c:281:11: error: dereferencing pointer to incomplete type
   ui.c:292:43: error: dereferencing pointer to incomplete type
   ui.c:295:13: error: dereferencing pointer to incomplete type
   ui.c:296:13: error: dereferencing pointer to incomplete type
   make[2]: *_\* [ui.o] Error 1
   make[2]: Leaving directory `/home/makefu/r/Honeyd/honeyd'
   make[1]: *** [all-recursive] Error 1
   make[1]: Leaving directory`/home/makefu/r/Honeyd/honeyd'
   make: **\* [all] Error 2

Thanks for the support.

I would love to see DataSoft continuing the really great honeyd project as an open-source, community supported project

**Comment by DataSoft** [comment]
Thanks!

Our build machines are Ubuntu, so it's got to be a dependency issue you're having. The libevent thing for sure. As of Ubuntu 11.04+, there's two versions of libevent available in the repos: "libevent-dev" and "libevent1-dev". The second one is what we need, or else you get build errors. I'm not positive what version of libevent is necessary for archlinux just because I haven't used it personally before. :) But I would keep a lookout for a similar package. 

btw, I updated the wiki here: https://github.com/DataSoft/Honeyd/wiki with our build process and dependencies. Again, it's written for Ubuntu, but it might shed some light.

All those warnings are expected, we get them too. So you can safely ignore them. One of the things we want to do is clean that up. You shouldn't "expect" compiler warnings. Expect an issue made for it soon. Once we're back from holiday break, there'll be more activity here.

The second build error also looks like a libevent erorr. It's complaining about dereferencing the buffer variable (who's type is incomplete), which is a pointer to a struct evbuffer.That struct is defined in libevent. So definitely the same issue there.

**Comment by worldwise001** [comment]
So I decided to see if I could track down this issue. Turns out Arch's libevent is too new (being >= 2). It works fine with libevent1 from AUR (http://aur.archlinux.org/packages.php?ID=47737). The only issue with this is that if you have anything that explicitly relies on libevent being >= 2, then this won't work, as libevent1 and libevent cannot co-exist. A full list of what requires libevent is here: http://www.archlinux.org/packages/core/x86_64/libevent/ OR http://www.archlinux.org/packages/core/i686/libevent/

Other relevant packages for honeyd: 
libdnet
libpcap
pcre
libedit
bison
flex
libtool
automake

Otherwise the package seems to make and install and run fine.

I might add a PKGBUILD for this at some point in the future.
